src/test/java/bio/terra/common/fixtures/ConnectedOperations.java

```diff
@@ -714,9 +754,15 @@ public MockHttpServletResponse validateJobModelAndWait(MvcResult inResult) throw
     // -- tracking methods --
 
     public void addDataset(String id) {
+        logger.info("Cleanup Tracking: Adding Dataset to list to be removed in cleanup. DatasetId: {}", id);
         createdDatasetIds.add(id);
     }
 
+    public void removeDatasetFromTracking(String id) {
+        logger.info("Cleanup Tracking: Removing Dataset from tracking list. DatasetId: {}", id);
+        createdDatasetIds.remove(id);
+    }
```

Comment:
@RoriCremer I know we had talked about doing something like this in one of your tests - basically, if we've already deleted a dataset, removing it from the list so that teardown won't try to delete it again. Did you already handle this in one of your PRs? Just don't want to step on your changes.